hummingbot/connector/exchange_base.pyx

```diff
@@ -168,9 +168,9 @@ cdef class ExchangeBase(ConnectorBase):
         cdef:
             OrderBook order_book = self.c_get_order_book(trading_pair)
             OrderBookQueryResult result = order_book.c_get_vwap_for_volume(is_buy, float(volume))
-            object query_volume = self.c_quantize_order_amount(trading_pair, Decimal(result.query_volume))
-            object result_price = self.c_quantize_order_price(trading_pair, Decimal(result.result_price))
-            object result_volume = self.c_quantize_order_amount(trading_pair, Decimal(result.result_volume))
+            object query_volume = Decimal(str(result.query_volume))
```

Comment:
I don't follow why we do this "blind" `str` conversion. I understand that the representation of an arbitrary human-defined float (i.e. a precise rational with a given precision) is perceived as wrong with say Decimal(0.03), thus requiring Decimal("0.03") - note that this is NOT str(0.03)  - but Decimal(str(float)) seems to be a blind and implicit formatting of the float. I think it would be much clearer to specify the precision that we expect with the conversion

Reply:
We could be specific, `str(float)` will vary depending on the machine you use (I think, it would, back in the days of Sun spark station double 16digits vs pentium 18digits  when I was writing a finite-element code). For instance, we could write `Decimal(str(round(float, 9)))` or `Decimal(f"{float:.9}")` - Anyway, likely knit-picking, I think the str() gives the illusion of infinite precision like in Decimal("0.03") == Decimal(str(0.03)), which I think is rarely true